controllers/providers/pxc/applier.go

```diff
@@ -126,6 +126,9 @@ func (p *applier) Engine() error {
 	if p.DB.Status.Status == everestv1alpha1.AppStateReady {
 		pxc.Spec.PXC.PodSpec.Affinity = p.currentPerconaXtraDBClusterSpec.PXC.Affinity
 	}
+
+	p.configureEngineAffinity()
```

Comment:
I think we need to do some cleanup above because we were setting some default affinity that we got from `common.DefaultAffinitySettings`.
Besides cleaning up those defaults, we should also ensure that we don't change the affinity for existing cluster (see the if condition 4 lines above this comment) otherwise all DB clusters would restart when upgrading the everest operator to 1.4.

Reply:
I'm not sure I follow.
Re restarts - the `p.configureEngineAffinity()` triggers a change only in case you configure affinity configuration for an `Engine`.
Since the spec does not support affinity at the moment, existing clusters shall not be affected.

Re cleanup - the line https://github.com/percona/everest-operator/blob/3a0cc19738d8fc4611a89e3be54c98433640e95c/controllers/providers/pxc/affinity.go#L24 cleans up previous affinity settings. Is there anything else you had in mind?

Reply:
You're right.
I thought it's better not to restart the DB and keep the behavior consistent across minor versions.
Happy to change this in a way which makes more sense. It doesn't seem to be possible to determine if the default affinity settings are expected if affinity is not provided.